dotnet/src/SemanticKernel/CoreSkills/TextMemorySkill.cs

```diff
@@ -119,30 +121,21 @@ public string Recall(string text, SKContext context)
         // TODO: support locales, e.g. "0.7" and "0,7" must both work
         var limitInt = int.Parse(limit, CultureInfo.InvariantCulture);
         var relevanceThreshold = float.Parse(relevance, CultureInfo.InvariantCulture);
-        var memories = context.Memory
-            .SearchAsync(collection, text, limitInt, relevanceThreshold)
-            .ToEnumerable();
 
-        context.Log.LogTrace("Done looking for memories in collection '{0}')", collection);
-
-        string resultString;
-
-        if (limitInt == 1)
-        {
-            var memory = memories.FirstOrDefault();
-            resultString = (memory != null) ? memory.Metadata.Text : string.Empty;
-        }
-        else
-        {
-            resultString = JsonSerializer.Serialize(memories.Select(x => x.Metadata.Text));
-        }
+        // Search memory
+        List<MemoryQueryResult>? memories = await context.Memory
```

Comment:
`memories` should never be null.  You can remove the `?` as well as the null check below.